recipe-server/normandy/recipes/models.py

```diff
@@ -477,15 +477,48 @@ def close(self):
         recipe.save()
 
 
+class ActionQuerySet(models.QuerySet):
+    def update_signatures(self):
```

Comment:
Should we wrap this in a transaction? That way we don't have partially-updated actions if something goes wrong.